sdk/Sdk/Tasks/ZipDeploy/ZipDeploymentStatus.cs

```diff
@@ -0,0 +1,133 @@
+            {
+                try
+                {
+                    deployStatus = await GetDeploymentStatusAsync(deploymentUrl, userName, password, RetryCount, TimeSpan.FromSeconds(RetryDelaySeconds), tokenSource);
+                    if (_logMessages)
+                    {
+                        _log.LogMessage(String.Format(StringMessages.DeploymentStatus, Enum.GetName(typeof(DeployStatus), deployStatus)));
+                    }
+                }
+                catch (HttpRequestException)
+                {
+                    return DeployStatus.Unknown;
+                }
+
+                await Task.Delay(TimeSpan.FromSeconds(StatusRefreshDelaySeconds));
+            }
+
+            return deployStatus;
+        }
+
+        private async Task<DeployStatus> GetDeploymentStatusAsync(string deploymentUrl, string userName, string password, int retryCount, TimeSpan retryDelay, CancellationTokenSource cts)
+        {
+            IDictionary<string, string>? json = await InvokeGetRequestWithRetryAsync<Dictionary<string, string>>(deploymentUrl, userName, password, retryCount, retryDelay, cts);
+            string? statusString = null;
+            if (json != null && !json.TryGetValue("status", out statusString))
+            {
+                return DeployStatus.Unknown;
+            }
+
+            if (Enum.TryParse(statusString, out DeployStatus result))
+            {
+                return result;
+            }
+
+            return DeployStatus.Unknown;
+        }
+
+        private async Task<T?> InvokeGetRequestWithRetryAsync<T>(string url, string userName, string password, int retryCount, TimeSpan retryDelay, CancellationTokenSource cts)
+        {
+            IHttpResponse? response = null;
+            await RetryAsync(async () =>
+            {
+                response = await _client.GetWithBasicAuthAsync(new Uri(url, UriKind.RelativeOrAbsolute), userName, password, _userAgent, cts.Token);
+            }, retryCount, retryDelay);
+
+            if (response!.StatusCode != HttpStatusCode.OK && response.StatusCode != HttpStatusCode.Accepted)
+            {
+                return default;
+            }
+            else
+            {
+                using (var stream = await response.GetResponseBodyAsync())
+                {
+                    return await JsonSerializer.DeserializeAsync<T>(stream);
+                }
+            }
+        }
+
+        private async Task RetryAsync(Func<Task> func, int retryCount, TimeSpan retryDelay)
+        {
+            while (true)
+            {
+                try
+                {
+                    await func();
+                    return;
+                }
+                catch (Exception e)
+                {
+                    if (retryCount <= 0)
+                    {
+                        throw e;
```

Comment:
Shouldn't this re-throw?

```suggestion
                        throw;
```

Reply:
Yes.